### ISSUE TITLE
🐛 Vue: False positive `parse` and `noAssignInExpressions` errors in `<template>` starting from `2.4.0`

### ISSUE THREAD

**Issue by astrochemx** [biomejs/biome#9161]
### Environment information

<details>

```bash

CLI:
  Version:                      2.4.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v25.6.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/11.0.0-alpha.5

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  true
  GraphQL enabled:              true
  Recommended:                  true
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noImportantStyles
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useGraphqlNamedOperations
    correctness/useHookAtTopLevel
    correctness/useIsNan
    correctness/useParseIntRadix
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/useSortedClasses
    performance/noAccumulatingSpread
    performance/noDynamicNamespaceImportAccess
    performance/noImgElement
    security/noBlankTarget
    security/noGlobalEval
    style/noDescendingSpecificity
    style/noNonNullAssertion
    style/useArrayLiterals
    style/useConst
    style/useDeprecatedReason
    style/useExponentiationOperator
    style/useExportType
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodejsImportProtocol
    style/useShorthandFunctionType
    style/useTemplate
    suspicious/noApproximativeNumericConstant
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConstEnum
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noExplicitAny
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUnusedExpressions
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword

Workspace:
  Open Documents:               0

```

</details>

### What happened?

Starting from Biome `2.4.0`, the parser and linter are producing false positives within Vue SFC templates:

1. `parse` **error**: Biome incorrectly attempts to parse attribute values containing spaces (UnoCSS/TailwindCSS class groupings) as single JavaScript expressions – `Template expressions can only contain a single expression`.

2. `noAssignInExpressions` **error**: Biome triggers a lint error for assignments within `@click` handlers, which are standard and valid in Vue template syntax – `The assignment should not be in an expression`.

This worked as expected in `2.3.15` and earlier Biome versions.

### Actual Behavior

<details>

<summary>Example Biome Output</summary>

```
components\Counter.vue:31:26 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × Template expressions can only contain a single expression

    29 │       font="mono"
    30 │       outline="!none"
  > 31 │       hover:bg="gray-400 opacity-20"
       │                          ^^^^^^^
    32 │       @click="counter += 1"
    33 │     >

  i Remove the extra code or split into multiple template expressions.


components\Counter.vue:32:15 lint/suspicious/noAssignInExpressions ━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × The assignment should not be in an expression.

    30 │       outline="!none"
    31 │       hover:bg="gray-400 opacity-20"
  > 32 │       @click="counter += 1"
       │               ^^^^^^^^^^^^
    33 │     >
    34 │       +

  i The use of assignments in expressions is confusing.
    Expressions are often considered as side-effect free.
```

</details>

### Steps to Reproduce

1. Create a `.vue` file with the following content:

```
<template>
  <button
    type = "button"
    hover:bg="gray-400 opacity-20"
    @click="counter += 1"
  >
    +
  </button>
</template>
```

2. Run `biome check` or `biome lint` using version `2.4.0` or later.

### Minimal Reproduction

[Playground Link](https://biomejs.dev/playground/?code=PAB0AGUAbQBwAGwAYQB0AGUAPgAKACAAIAA8AGIAdQB0AHQAbwBuAAoAIAAgACAAIAB0AHkAcABlACAAPQAgACIAYgB1AHQAdABvAG4AIgAKACAAIAAgACAAaABvAHYAZQByADoAYgBnAD0AIgBnAHIAYQB5AC0ANAAwADAAIABvAHAAYQBjAGkAdAB5AC0AMgAwACIACgAgACAAIAAgAEAAYwBsAGkAYwBrAD0AIgBjAG8AdQBuAHQAZQByACAAKwA9ACAAMQAiAAoAIAAgAD4ACgAgACAAIAAgACsACgAgACAAPAAvAGIAdQB0AHQAbwBuAD4ACgA8AC8AdABlAG0AcABsAGEAdABlAD4ACgA%3D&language=vue)

### Expected result

There should be no errors.

1. `hover:bg="gray-400 opacity-20"` should be treated as a static attribute string, not a template expression.
2. `@click="counter += 1"` should be recognized as a valid Vue event handler assignment and should not trigger `noAssignInExpressions`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
The `noAssignInExpressions` diagnostic is arguably not a false positive, but maybe we should consider having the rule ignore this case because its a rather common practice.

The parser error is a bug though. The bug is that it thinks `hover` is it's own attribute, and as a result parses `:bg` as a v-bind shorthand.

 Just curious, is `hover:bg` special syntax for some library? AFAIK, the way you use tailwind in vue is to put things in the `class` attribute

**Comment by astrochemx** [comment]
Thanks for the quick look and clarification!
First of all, I am not a Vue expert myself, so I’m answering based on my basic knowledge.

**Regarding `noAssignInExpressions`:**

That makes sense – I agree it’s not technically a false positive from a JavaScript perspective. However, it would be great if this rule could be silenced specifically for `v-on` / `@` attributes. These are explicitly designated as ***inline event handlers***, where side effects are expected and idiomatic. Suppressing the rule in that specific context would likely align better with real-world Vue usage.

**Regarding the `hover:bg` syntax:**

Yes, this is a special (but standard) syntax used by **UnoCSS** (specifically the [Attributify Preset](https://unocss.dev/presets/attributify)). It allows writing utilities as attributes rather than long strings inside the `class` attribute.

In **attributify mode**, this:

```
hover:bg="gray-400 opacity-20"
```

is essentially a shorthand for:

```
class="hover:bg-gray-400 hover:bg-opacity-20"
```

In Vue, a colon is only treated as a `v-bind` shorthand if it is the prefix of the attribute (e.g., `:bg="..."`). When it appears in the middle of the name (like `hover:bg`), it is meant to be parsed as a literal attribute name.
